### PR TITLE
Apply elevation scale in components package

### PR DESCRIPTION
**Pull request by jameskoster**
Elevation

Follow up to https://github.com/WordPress/gutenberg/pull/64108

## What?
This PR applies the recently-added elevation scale to components in the core package.

> [!IMPORTANT]  
> This PR gives visibility to new steps in the elevation scale for the first time, most notably `medium` elevation applied to menus / popovers. If we're unhappy with the appearance, or even where certain components are placed, now would be a good time to adjust. For me, `medium` elevation probably needs to be toned down a bit.

## Why?
See https://github.com/WordPress/gutenberg/issues/63757

### Custom Select Control v2
Applied `medium` elevation

| Before | After |
| --- | --- |
| Screenshot 2024-09-09 at 16 46 11 | Screenshot 2024-09-17 at 09 54 26 |

### DropdownMenu v2
Applied `medium` elevation

| Before | After |
| --- | --- |
| Screenshot 2024-09-09 at 16 47 32 | Screenshot 2024-09-17 at 09 55 12 |

### Form Toggle
Applied `x-small` elevation to handle

| Before | After |
| --- | --- |
| Screenshot 2024-09-09 at 16 48 56 | Screenshot 2024-09-09 at 16 48 59 |

### Popover
Applied `medium` elevation

| Before | After |
| --- | --- |
| Screenshot 2024-09-09 at 16 50 20 | Screenshot 2024-09-17 at 09 55 46 |

### Range Control
Applied `x-small` elevation to handle

| Before | After |
| --- | --- |
| Screenshot 2024-09-09 at 16 51 23 | Screenshot 2024-09-09 at 16 51 29 |

### Resizeable Box
Applied `x-small` elevation to handles

| Before | After |
| --- | --- |
| Screenshot 2024-09-09 at 16 52 16 | Screenshot 2024-09-17 at 09 56 17 |

### Snackbar
Applied `small` elevation

| Before | After |
| --- | --- |
| Screenshot 2024-09-09 at 16 53 13 | Screenshot 2024-09-12 at 22 48 33 |

### Tooltip
Applied `small` elevation

| Before | After |
| --- | --- |
| Screenshot 2024-09-09 at 16 54 13 | Screenshot 2024-09-12 at 22 49 24 |

## Testing Instructions
1. `npm run storybook:dev`
2. Check the components detailed above appear as expected
3. Test the Editor

> [!NOTE]  
> This PR does not touch the `Elevation` component. So far as I can tell it is not frequently used – it might be good to discuss it's long-term future.

### CODE REVIEW

**Review by ciampo — COMMENTED** [review]
On a quick look, code changes look good — we mostly need to review the visual impact of those changes.

> medium elevation probably needs to be toned down a bit.

That's also my impression

**Comment by jameskoster** [comment]
I'm going to apply the values from [Figma](https://www.figma.com/design/804HN2REV2iap2ytjRQ055/Core-System-Library?node-id=8760-33554&t=taFNGW4tpLeUn2Pz-1) and see how that goes :)

**Comment by jameskoster** [comment]
> I'm going to apply the values from [Figma](https://www.figma.com/design/804HN2REV2iap2ytjRQ055/Core-System-Library?node-id=8760-33554&t=taFNGW4tpLeUn2Pz-1) and see how that goes :)

Done in https://github.com/WordPress/gutenberg/pull/65159/commits/248f1e6c75fa5e32bec08d1bcc5a5c011eee9c1f. 

Screenshot 2024-09-10 at 18 32 51


I think it's a step in the right direction 💪 I'll update the screenshots in the OP when we're happy with the values.

**Comment by jasmussen** [comment]
Separate from the work, but possibly relevant when tuning any values, it'd be nice to make the scrim itself a bit lighter.

**Comment by jameskoster** [comment]
> Separate from the work, but possibly relevant when tuning any values, it'd be nice to make the scrim itself a bit lighter.

Let's try that separately as it's part of the Modal component. 

This PR is about applying elevation (shadows) to the components listed in the OP, according to the scale defined in https://github.com/WordPress/gutenberg/pull/64108.

**Review by tyxla — APPROVED** [review]
The code looks good to me. I also did some testing and the new elevation looks great. 👍 

It did feel a bit intense in some places with popovers, but I guess that's just because I'm not used to it. 😅

**Comment by tyxla** [comment]
BTW, let's add a CHANGELOG entry. This is definitely worth mentioning.

File: packages/base-styles/_variables.scss [hunk 1]
```diff
@@ -55,16 +55,16 @@ $radius-round: 50%;     // For circles and ovals.
...
-$elevation-small: 0 0.7px 1px 0 rgba(0, 0, 0, 0.12), 0 2.2px 3.7px -0.2px rgba(0, 0, 0, 0.12), 0 5.3px 7.3px -0.5px rgba(0, 0, 0, 0.12);
+$elevation-small: 0 1px 2px rgba(0,0,0, 0.05), 0 2px 3px rgba(0,0,0, 0.04), 0 6px 6px rgba(0,0,0, 0.03), 0 8px 8px rgba(0,0,0, 0.02);
 
 // For components that offer additional actions. Example: Menus, Command Palette
-$elevation-medium: 0 0.7px 1px 0 rgba(0, 0, 0, 0.14), 0 4.2px 5.7px -0.2px rgba(0, 0, 0, 0.14), 0 7.3px 9.3px -0.5px rgba(0, 0, 0, 0.14);
+$elevation-medium: 0 2px 3px rgba(0,0,0, 0.05), 0 4px 5px rgba(0,0,0, 0.04), 0 12px 12px rgba(0,0,0, 0.03), 0 16px 16px rgba(0,0,0, 0.02);
 
 // For components that confirm decisions or handle necessary interruptions. Example: Modals.
-$elevation-large: 0 0.7px 1px rgba($black, 0.15), 0 2.7px 3.8px -0.2px rgba($black, 0.15), 0 5.5px 7.8px -0.3px rgba($black, 0.15), 0.1px 11.5px 16.4px -0.5px rgba($black, 0.15);
+$elevation-large: 0 5px 15px rgba(0,0,0, 0.08), 0 15px 27px rgba(0,0,0, 0.07), 0 30px 36px rgba(0,0,0, 0.04), 0 50px 43px rgba(0,0,0, 0.02);
```
Comment (ciampo) [inline packages/base-styles/_variables.scss]: The lack of spaces after the commas is causing a lint error.


```suggestion
$elevation-x-small: 0 1px 1px rgba(0, 0, 0, 0.03), 0 1px 2px rgba(0, 0, 0, 0.02), 0 3px 3px rgba(0, 0, 0, 0.02), 0 4px 4px rgba(0, 0, 0, 0.01);

// For components that provide contextual feedback without being intrusive. Generally non-interruptive. Example: Tooltips, Snackbar.
$elevation-small: 0 1px 2px rgba(0, 0, 0, 0.05), 0 2px 3px rgba(0, 0, 0, 0.04), 0 6px 6px rgba(0, 0, 0, 0.03), 0 8px 8px rgba(0, 0, 0, 0.02);

// For components that offer additional actions. Example: Menus, Command Palette
$elevation-medium: 0 2px 3px rgba(0, 0, 0, 0.05), 0 4px 5px rgba(0, 0, 0, 0.04), 0 12px 12px rgba(0, 0, 0, 0.03), 0 16px 16px rgba(0, 0, 0, 0.02);

// For components that confirm decisions or handle necessary interruptions. Example: Modals.
$elevation-large: 0 5px 15px rgba(0, 0, 0, 0.08), 0 15px 27px rgba(0, 0, 0, 0.07), 0 30px 36px rgba(0, 0, 0, 0.04), 0 50px 43px rgba(0, 0, 0, 0.02);
```

File: packages/base-styles/_variables.scss [hunk 2]
```diff
@@ -55,16 +55,16 @@ $radius-round: 50%;     // For circles and ovals.
  */
 
 // For sections and containers that group related content and controls, which may overlap other content. Example: Preview Frame.
-$elevation-x-small: 0 0.7px 1px rgba($black, 0.1), 0 1.2px 1.7px -0.2px rgba($black, 0.1), 0 2.3px 3.3px -0.5px rgba($black, 0.1);
+$elevation-x-small: 0 1px 1px rgba(0,0,0, 0.03), 0 1px 2px rgba(0,0,0, 0.02), 0 3px 3px rgba(0,0,0, 0.02), 0 4px 4px rgba(0,0,0, 0.01);
```
Comment (ciampo) [inline packages/base-styles/_variables.scss]: What's the reason for switching from `$black` to `0, 0, 0` ?

Comment (jameskoster) [reply packages/base-styles/_variables.scss]: Oh, I was copy / pasting between `_variables.scss` and `config-values.js`. `$black` doesn't work in the latter :)

I'll revert to `$black` once we're happy with the values.

**Comment by ciampo** [comment]
Should we go ahead with the current values (after [switching back to `$black`](https://github.com/WordPress/gutenberg/pull/65159#discussion_r1761694904)) and follow-up if necessary?

**Comment by jameskoster** [comment]
It'd be good to get feedback from someone in @WordPress/gutenberg-design . But if we don't hear anything in the next day or so then yes. It's easy to revisit the values if necessary.

**Review by keoshi — APPROVED** [review]
The changes look great and definitely add a more tactile touch to these elements!

Unrelated to this PR, but I had to isolate the shadow in these two elements to check if my eyes weren't deceiving me.

image

The much darker border makes the shadow in "Custom Select Control v2" look a lot more subtle and even anatomically different, even though they share the same `box-shadow` attributes.

**Comment by jameskoster** [comment]
Thank for the review @keoshi. 

I agree, the dark border in Custom Select Control V2 is definitely a bit off visually. We should consider swapping that to `$gray-300` to align with menus and popovers. Generally the dark (`$gray-900`) border is reserved only for the block toolbar and extensions thereof.

**Comment by ciampo** [comment]
FYI, `CustomSelectControl V2` is not really ready for usage, so I wouldn't worry too much about how it looks at this point in time. Although the feedback applies to `DropdownMenu`, `DropdownMenuV2` and `Popover`.

**Comment by crisbusquets** [comment]
Works for me! 🚀 Thanks for working on this @jameskoster !

I like how it's now easier to see the layering, where the popover rises above the content, improving the overall hierarchy. :smile:

**Comment by jameskoster** [comment]
Thanks all. Let's merge this one, we can easily revisit the values later if we need to :)

**Pull request merged**